### PR TITLE
[css-to-js] Invalid JS when mixing rules with composes

### PR DESCRIPTION
**Pull request by plesiecki**
## Description

Hello 👋 I've spotted an issue with `css-to-js` module. [For CSS](https://github.com/plesiecki/modular-css/blob/df11a5be29fc66516de7ae60836a9adf515f61cd/packages/css-to-js/test/specimens/composes/custom-between-rules.css#L1-L9):
```css
.rule { 
  display: flex;
}
.custom {
  color: pink;
}
.rule {
  composes: custom;
}
```
it produces an [invalid JS](https://github.com/plesiecki/modular-css/blob/df11a5be29fc66516de7ae60836a9adf515f61cd/packages/css-to-js/test/__snapshots__/api.test.js.snap#L61) which looks like this:
```js
const rule =  + " " + "mc9c938a76_rule"
```

## Motivation and Context
A valid CSS can give an invalid JS.

## How Has This Been Tested?
Unit tests have been updated.

**For the second test a snapshot have been updated manually so the test should fail.**

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by plesiecki** [comment]
It works for 81fd1bbac239c69a9adaa0f3dcc2874df6090b68, @tivac what do you think about that change?

File: packages/css-to-js/test/api.test.js [hunk 1]
```diff
@@ -168,6 +165,27 @@ describe("@modular-css/css-to-js API", () => {
...
+        const processor = new Processor({ resolvers });
+
+        await processor.file(require.resolve("./specimens/composes/custom-first-rule.css"));
+
+        const { code } = transform(require.resolve("./specimens/composes/custom-first-rule.css"), processor);
+
+        expect(code).toMatchSnapshot("code");
+    });
+
+    it("should generate javascript from multiple composes", async () => {
```
Comment (KubaZ) [inline packages/css-to-js/test/api.test.js:179]: will local composes declaration work, if referenced class is defined after usage?

Comment (plesiecki) [reply packages/css-to-js/test/api.test.js:179]: If you mean something like [this](https://m-css.com/repl/#NrBEHoFsEMEsDsB0kDGBnNoA0pEDMB7AgAgG8AdeY64lAyABwLQFM0AuYgI2gCcBuSgF9KlRD15lKNWgQA2BXp14sAJoPhDQAXW1A) then in such a case the `processor` throws
https://github.com/tivac/modular-css/blob/e64362e087f4ae6113a1b622e59dab827a9f2e97/packages/processor/plugins/composition.js#L66-L71

Comment (tivac) [reply packages/css-to-js/test/api.test.js:179]: When we switched to the new walking APIs in `postcss@8` it meant that out-of-order declarations would've been a bunch of extra work to support, so I chose not too.

Doesn't mean you couldn't support them using a "gather and process" approach, and for at least one type we do that, but given how `composes` can be several levels deep it'd require some real careful planning.

**Review by tivac — COMMENTED** [review]
So the issue being fixed here is that when we walked the exported keys sometimes we were walking classes out of dependency order, and your fix is to always walk all the local classes first and ensure they've had a JS identifier created before actually creating the value for them?

**Comment by plesiecki** [comment]
> So the issue being fixed here is that when we walked the exported keys sometimes we were walking classes out of dependency order, and your fix is to always walk all the local classes first and ensure they've had a JS identifier created before actually creating the value for them?

Basically yes.

**Pull request merged**